### PR TITLE
Add ping command to test for successful connection

### LINKED ISSUE #24: Test database connection - CLI
## Description

There needs to be a command to test the connection to a database for ensuring that the setup was correct. 

## Other notes

This should be a CLI command e.g. `monosi test-connection` or `monosi ping`  or something similar.

### PR DESCRIPTION
**Pull request by annamarieweber**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Other (please describe):

## Issue

resolves #24

## Description of change & new behavior

Adds CLI command 'ping' which takes an optional second argument source
and will test the connection for the specified source. If no source is
specified 'default' is used

### CODE REVIEW

File: src/cli/cmd/ping.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+          driver_config = self.config.sources[source]
+          driver_cls = load_driver(driver_config)
+          driver = driver_cls(driver_config)
+          driver.test()
+      except DBAPIError as err:
+          print(f"Error Connecting to Data Source {err=}")
+      except OperationalError as err:
+          print(f"Invalid Source Configuration {err=}")
+      except KeyError as err:
+          print(f"Source \'{source}\' was not found. Must use one of: {list(self.config.sources.keys())}")
```
Comment (annamarieweber) [inline src/cli/cmd/ping.py:24]: wasn't sure if I should follow the pattern used in profile, etc and delegate this to a task just calling the tasks from here or of that was getting too into the weeds, went back and forth but I figure this is a start.

File: src/core/common/drivers/base.py [hunk 2]
```diff
@@ -153,6 +153,12 @@ def close(self):
             self.connection.close()
         self.connection = None
 
+    def test(self):
```
Comment (unkrich) [inline src/core/common/drivers/base.py:156]: This doesn't actually test the connection. It tests whether we have previously tried to run a query successfully. 

We should be running an actual query here as a test and verifying the response. E.g. SELECT 1;

File: src/cli/__init__.py [hunk 3]
```diff
@@ -52,6 +56,11 @@ def profile(self, args):
         args = None
         task = ProfileCmd.from_args(args)
         task.run()
+    
+    def ping(self, args):
+        args = args[1] if len(args)>1  else "default"
```
Comment (unkrich) [inline src/cli/__init__.py:61]: args should be based directly and used in the cmd as in line 39

File: src/cli/cmd/ping.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+from sqlalchemy.exc import OperationalError,DBAPIError
+class PingCmd(BaseCmd):
+  def run(self):
+      source = self.args
+      try:
+          driver_config = self.config.sources[source]
+          driver_cls = load_driver(driver_config)
+          driver = driver_cls(driver_config)
+          driver.test()
```
Comment (unkrich) [inline src/cli/cmd/ping.py:18]: This should output the result and the status of the test should be displayed accordingly

File: src/cli/cmd/ping.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      except DBAPIError as err:
```
Comment (unkrich) [inline src/cli/cmd/ping.py:19]: Likely sqlalchemy error imports should be imported and handled directly in the driver class

File: src/cli/cmd/ping.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+from cli.utils.yaml import write_file
+ 
+from core.common.drivers.factory import load_driver
+from core.common.drivers.factory import load_config
+from .base import BaseCmd
+
+from sqlalchemy.exc import OperationalError,DBAPIError
+class PingCmd(BaseCmd):
+  def run(self):
+      source = self.args
```
Comment (unkrich) [inline src/cli/cmd/ping.py:13]: Consider checking for source attribute and then assigning accordingly. 

Logging.warn if none provided that we're using default

File: src/cli/cmd/ping.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+from cli.config import ProjectConfiguration
+from cli.utils.yaml import write_file
+ 
+from core.common.drivers.factory import load_driver
+from core.common.drivers.factory import load_config
+from .base import BaseCmd
+
+from sqlalchemy.exc import OperationalError,DBAPIError
+class PingCmd(BaseCmd):
+  def run(self):
```
Comment (unkrich) [inline src/cli/cmd/ping.py:12]: No output if the test was successful

File: src/cli/cmd/ping.py (hunk 1 above)
Comment (unkrich) [reply src/cli/cmd/ping.py:24]: Profile command is a mess currently.

**Review by unkrich — COMMENTED** [review]
This is a great start! Thanks for the contribution. 

I added a couple comments to the PR - would love your thoughts.

**Pull request closed without merging**